awscli/customizations/cloudformation/artifact_exporter.py

```diff
@@ -94,6 +94,55 @@ def parse_s3_url(url,
                      "{0}".format(url))
 
 
+def get_nested_property_value(resource_dict, property_name):
+    """
+    Searches the resource_dict for nested properties by splitting the property_name
+    on '.'
+
+    :param resource_dict:   Dictionary containing resource definition
+    :param property_name:   Property name of CloudFormation resource where this
+                            local path is present
+
+    :return:                Value of the property
+    """
+
+    # Support nested properties by allowing '.' in the PROPERTY_NAME
+    if '.' in property_name:
```

Comment:
Have you thought about doing some recursion here instead? Seems like a better way forward and would remove the explicit if else branch on "if . is in the string". 

You can use `property_name.split('.', 1)`. The base case would be the list with 1 item (might need a `None` check). I find the nest if's with loops like this generally harder to read.

Reply:
Thanks @jfuss.  I've updated the get_nested_property_value and set_nested_property_value to use recursion instead of the nested for/if structure.